mavlink temporary workarounds for dronekit

RomanBapst opened the pull request:
Temporary workaround for companion systems which rely on dronekit.
See commit message for further explanation.
@sanderux Could you review and also correct me if I haven't explained the motivation well enough?

dagar commented on src/modules/mavlink/mavlink_params.c:
```diff
@@ -141,3 +141,27 @@ PARAM_DEFINE_INT32(MAV_FWDEXTSP, 1);
+ * Disabling the parameter hash check functionality will make the mavlink instance
+ * stream parameters continuously.
+ *
+ * @value 0 Hash check enabled.
```
`@boolean`

dagar replied:
Actually, that probably only makes sense if you want to invert the logic to be parameters that enable.

dagar commented on src/modules/mavlink/mavlink_params.c:
```diff
@@ -141,3 +141,27 @@ PARAM_DEFINE_INT32(MAV_FWDEXTSP, 1);
+ * The mavlink hearbeat message will not be forwarded if this parameter is set to 'disabled'.
+ * The main reason for disabling heartbeats to be forwarded is because they confuse dronekit.
+ *
+ * @value 0 Forwarding enabled.
```
`@boolean`

dagar commented:
Overall looks fine, although you might want to consider using the Param wrapper for this kind of thing. It's fewer moving pieces and provides static checks. https://github.com/PX4/Firmware/blob/master/src/platforms/px4_param.h

RomanBapst commented:
@dagar I addressed your comments except the one regarding the parameter wrapper. Are you suggesting to change the mavlink module to use ModuleParams? If this is what you want then I can do it for all the others parameters as well not just for the ones I added.

dagar commented:
Yes I'm suggesting to change to ModuleParams (when convenient).

RomanBapst commented:
@dagar Ok, will do that then.

RomanBapst commented:
Closing in favor of #10422
